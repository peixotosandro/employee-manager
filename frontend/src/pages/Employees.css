section#employee-list {
  width: 100%;
  max-width: 900px;
  margin: 0 auto;
}

section#employee-list .row {
  padding: 5px 0;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #ddd;
}

section#employee-list .col span {
  font-size: 16px;
}
