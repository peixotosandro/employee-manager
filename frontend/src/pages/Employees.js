import React, { Component } from "react";
import io from "socket.io-client";
import Dialog from "react-bootstrap-dialog";
import { Button, Container, Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";

import api from "../services/api";
import "./Employees.css";

class Employees extends Component {
  state = {
    employees: []
  };

  async componentDidMount() {
    this.listenToSocket();

    try {
      const response = await api.request.get("employee");

      this.setState({ employees: response.data });
    } catch (error) {
      if (error.response) {
        this.handleError(error.response.data.error);
      } else if (error.request) {
        this.handleError("ERRO: Sem conexão com o servidor.");
      }
    }

    Dialog.setOptions({
      defaultCancelLabel: "Não"
    });
  }

  listenToSocket = () => {
    const socket = io(api.baseURL);

    socket.on("employeeRegistered", employee => {
      this.setState({
        employees: [...this.state.employees, employee]
      });
    });

    socket.on("employeeDeleted", id => {
      this.setState({
        employees: this.state.employees.filter(employee => {
          return id !== employee.id;
        })
      });
    });

    socket.on("employeeUpdated", async id => {
      const response = await api.request.get(`employee/${id}`);

      this.setState({
        employees: this.state.employees.map(employee => {
          if (id === employee.id) {
            employee = response.data;
          }
          return employee;
        })
      });
    });
  };

  handleDelete = (id, name) => {
    this.dialog.show({
      title: `Deletar ${name}`,
      body: "Você tem certeza?",
      actions: [
        Dialog.CancelAction(),
        Dialog.Action(
          "Sim",
          async () => {
            try {
              await api.request.delete(`/employee/${id}`);
            } catch (error) {
              if (error.response) {
                this.handleError(error.response.data.error);
              } else if (error.request) {
                this.handleError("ERRO: Sem conexão com o servidor.");
              }
            }
          },
          "btn-danger"
        )
      ],
      bsSize: "small",
      onHide: dialog => {
        dialog.hide();
      }
    });
  };

  handleError = error => {
    this.dialog.showAlert(error);
  };

  render() {
    return (
      <section id="employee-list">
        <Dialog
          ref={component => {
            this.dialog = component;
          }}
        />
        <Container>
          {this.state.employees.map(employee => (
            <Row key={employee.id}>
              <Col sm={3}>
                <span>{employee.name}</span>
              </Col>
              <Col sm={3}>
                <span>{employee.email}</span>
              </Col>
              <Col sm={3}>
                <span>{employee.departament}</span>
              </Col>
              <Col sm={2}>
                <Button
                  variant="danger"
                  onClick={() => this.handleDelete(employee.id, employee.name)}
                >
                  Deletar
                </Button>
              </Col>
              <Col sm={1}>
                <Link to={`/edit/${employee.id}`}>
                  <Button variant="success">Alterar</Button>
                </Link>
              </Col>
            </Row>
          ))}
        </Container>
      </section>
    );
  }
}

export default Employees;
