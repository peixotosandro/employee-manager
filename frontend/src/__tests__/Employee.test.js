import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Employees from "../pages/Employees";

configure({ adapter: new Adapter() });

describe("Testing Employees Component", () => {
  it("should render properly.", () => {
    const wrapper = shallow(<Employees />);

    expect(wrapper).toMatchSnapshot();
  });
});
