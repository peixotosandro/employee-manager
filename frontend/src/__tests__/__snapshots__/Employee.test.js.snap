// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Employees Component should render properly. 1`] = `ShallowWrapper {}`;
